app/controllers/concerns/foreman/controller/registration_commands.rb

```diff
@@ -16,7 +16,27 @@ def registration_args
   end
 
   def insecure
```

Comment:
Little refactoring with safe guard:

```ruby
return '' unless registration_params['insecure']&.presence

if download_utility == 'curl'
  '--insecure'
else
  '--no-check-certificate'
end
```

Reply:
Thanks, this is a nice way to check the `insecure` flag. But regarding to some tests I did, the first line has to be `return '' unless registration_params['insecure']`. With this little change the results I get are

- `''` in case `insecure` is not set in the `registration_params`,
- `''` in case `insecure` is present in the `registration_params` an set to `false`, and
- in case  `insecure` is present in the `registration_params` an set to `true` the lines below are evaluated.

With the `&.presence` part I only get a `undefined method` error in case `insecure` is present in the `registration_params`.